nuxeo-core/nuxeo-core-storage/src/main/java/org/nuxeo/ecm/core/storage/FulltextExtractorWork.java

```diff
@@ -304,16 +293,23 @@ protected String blobToText(Blob blob) {
         }
     }
 
-    @SuppressWarnings("boxing")
-    protected String limitStringSize(String string, int maxSize) {
-        if (maxSize != 0 && string.length() > maxSize) {
-            if (log.isDebugEnabled()) {
-                log.debug(String.format("Fulltext extract of length: %s for document: %s truncated to length: %s",
-                        string.length(), docId, maxSize));
+    protected <O> String joinText(List<O> objects, Function<O, String> extractor, int maxSize) {
+        boolean limit = maxSize > 0;
+        // add space at beginning and end for simulated phrase search using LIKE "% foo bar %"
+        StringBuilder builder = new StringBuilder(" ");
+        int size = maxSize - 1;
+        for (O object : objects) {
+            String string = extractor.apply(object);
+            size -= string.length();
+            if (limit && size <= 0) {
+                builder.append(string, 0, string.length() + size);
```

Comment:
note that the last token is invalid when the string is truncated, at some point, we may want to remove it to avoid noise in search.

Reply:
Indeed, we thought about it but dismiss improving this code as it would make the change complex to understand.